tree/treeplayer/inc/ROOT/TDFInterface.hxx

```diff
@@ -364,6 +364,27 @@ public:
       TInterface<TTraits::TakeFirstParameter_t<decltype(upcastNode)>> upcastInterface(fProxiedPtr, fImplWeakPtr);
       // build a string equivalent to
       // "(TInterface<nodetype*>*)(this)->Snapshot<Ts...>(treename,filename,*(ColumnNames_t*)(&columnList))"
+
+      // The interpretation has multiple steps including
+      // 1. Parse
+      // 2. Codegen
+      // 3. Link
+      // 4. run static initializers
+      // 5. run user code
+      // 6. run static destructors.
+      //
+      // Where 1,2,3 explicitly modify the state of the (global) interpreter and must be (and, of course, are)
+      // covered by the ROOT global lock.  Technically, cling should release the lock when executing user
+      // code at step 4,5,6 but currently does not.  Since the user code can create threads that may
```

Comment:
Yes and the challenge is to find a way to release the lock slightly after the dlopen started (otherwise we must make a rule that no code run as part of the initialization must start thread that take the ROOT main lock).